base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp

```diff
@@ -176,6 +125,82 @@ bool PotentiallyHostArtifactsPath(const std::string& host_artifacts_path) {
                         std::back_inserter(result));
   return !result.empty();
 }
+
+Result<void> UpdateWebrtcDeviceId(cvd_common::Args& args,
+                                  const selector::LocalInstanceGroup& group) {
+  std::string flag_value;
+  std::vector<Flag> webrtc_device_id_flag{
+      GflagsCompatFlag("webrtc_device_id", flag_value)};
+  // ConsumeFlags modifies args, a copy is needed because only the presence
+  // of the flag is being tested.
+  std::vector<std::string> copied_args{args};
+  CF_EXPECT(ConsumeFlags(webrtc_device_id_flag, copied_args));
+
+  if (!flag_value.empty()) {
+    return {};
+  }
+
+  CF_EXPECT(!group.GroupName().empty());
+  std::vector<std::string> device_name_list;
+  for (const auto& instance : group.Instances()) {
+    std::string device_name =
+        group.GroupName() + "-" + instance.PerInstanceName();
+    device_name_list.push_back(device_name);
+  }
+  // take --webrtc_device_id flag away
```

Comment:
This comment is no longer accurate. The original flag is not being removed anymore, because `copied_args` is distinct and no longer being returned. Now the flag is potentially overridden / duplicated with the final value taking precedence.

Like the original, there is also probably a bad interaction here with `--num_instances` and a comma-separated-value being passed into `webrtc_device_id`.